File: tests/dynamic_fixtures/go/fileio_unsupported.go
// File I/O — unsupported fixture.
// Entry is a method on a struct.
// Test sets confidence = Low to get Unsupported(ConfidenceTooLow).
// Entry: FileServer.Serve  Cap: FILE_IO
// Expected verdict: Unsupported

package entry

import (
	"fmt"
	"os"
)

type FileServer struct{ BaseDir string }

func (s *FileServer) Serve(path string) {
	data, err := os.ReadFile(s.BaseDir + "/" + path)
	if err == nil {
		fmt.Print(string(data))
	}
}

File: tests/dynamic_fixtures/php/fileio_unsupported.php
<?php
// File I/O — unsupported fixture.
// Entry is a class method; test sets confidence = Low.
// Expected verdict: Unsupported

class FileServer {
    public function serve($path) {
        $content = @file_get_contents($path);
        if ($content !== false) {
            echo $content;
        }
    }
}
